src/Components/Publishing/Display/__test__/DisplayPanel.test.js

```diff
@@ -50,16 +65,108 @@ describe('units', () => {
+
+      return {
+        wrapper,
+        instance,
+        spy
+      }
+    }
+
+    it('tracks one time on mount', () => {
+      const wrapper = getWrapper()
+      const instance = wrapper.instance()
+      const spy = jest.spyOn(wrapper.props().tracking, 'trackEvent')
+      expect(spy).toHaveBeenCalledWith(expect.objectContaining({
+        action: 'Impression',
+        entity_type: 'display_ad',
+        unit_layout: 'panel'
+      }))
+    })
+
+    it('tracks video progress', () => {
+      const wrapper = getWrapper({ isVideo: true })
+      const instance = wrapper.instance()
+      const durationSpy = jest.spyOn(instance, 'trackDuration')
+      const secondsSpy = jest.spyOn(instance, 'trackSeconds')
+      instance.video.currentTime = 3
+      instance.trackProgress()
+      expect(durationSpy).toHaveBeenCalled()
+      expect(secondsSpy).toHaveBeenCalled()
+    })
+
+    it('tracks duration', () => {
+      const wrapper = getWrapper()
+      const instance = wrapper.instance()
+      const spy = jest.spyOn(wrapper.props().tracking, 'trackEvent')
+      instance.trackDuration(20)
+      expect(spy).toHaveBeenCalledWith(expect.objectContaining({
+        action: 'Video duration',
+        percent_complete: 20
+      }))
+    })
+
+    it('tracks seconds', () => {
+      const wrapper = getWrapper()
+      const instance = wrapper.instance()
+      const spy = jest.spyOn(wrapper.props().tracking, 'trackEvent')
+      instance.trackSeconds(20)
+      expect(spy).toHaveBeenCalledWith(expect.objectContaining({
+        action: 'Video seconds',
+        seconds_complete: 20
+      }))
+    })
+
+    it('tracks on click', () => {
+      const wrapper = getWrapper()
+      const instance = wrapper.instance()
+      const spy = jest.spyOn(wrapper.props().tracking, 'trackEvent')
+      const event = {
+        preventDefault: jest.fn(),
+        target: {
+          className: 'PlayButton'
+        }
+      }
+      instance.handleClick(event)
+      expect(spy).toHaveBeenCalledWith(expect.objectContaining({
+        action: 'Click',
+        label: 'Display ad clickthrough'
+      }))
     })
 
+    it('tracks on video click', () => {
+      const wrapper = getWrapper()
+      const instance = wrapper.instance()
+      const spy = jest.spyOn(wrapper.props().tracking, 'trackEvent')
+      instance.trackVideoClick()
+      expect(spy).toHaveBeenCalledWith(expect.objectContaining({
+        action: 'Click',
+        label: 'Display ad play video'
+      }))
+    })
+
+    it('tracks on mouseEnter, if desktop', () => {
+      const wrapper = getWrapper({ isVideo: true })
+      const instance = wrapper.instance()
+      const spy = jest.spyOn(wrapper.props().tracking, 'trackEvent')
+      instance.handleMouseEnter()
+      expect(spy).toHaveBeenCalledWith(expect.objectContaining({
+        action: 'MouseEnter',
+        label: 'Display ad play video'
+      }))
+    })
+  })
+
+  describe('#componentDidMount', () => {
```

Comment:
Lovely tests 🌷

Reply:
`@decorators` / `@track` is hard to test! Since it injects `props.tracking.trackEvent` anyways just decided to go with that rather than pulling my hair out.

Reply:
👍  never liked decorators myself anyways 🔥